pkg/api/message/v1/service.go

```diff
@@ -217,11 +217,15 @@ func (s *Service) Subscribe(req *proto.SubscribeRequest, stream proto.MessageApi
 	select {
 	case <-stream.Context().Done():
 		log.Debug("stream closed")
-		return nil
+		break
 	case <-s.ctx.Done():
 		log.Info("service closed")
-		return nil
+		break
 	}
+
+	metrics.EmitUnsubscribeTopics(stream.Context(), log, len(req.ContentTopics))
```

Comment:
Don't we want this in a `defer`, actually. Otherwise an error will increment the subscription but not the unsubscribe.